### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by ADi2991**
Sorry for the late submission. Won't happen again.
Aditya Rao (47221503)

### CODE REVIEW

File: hangman/lib/game/game.ex [hunk 1]
```diff
@@ -0,0 +1,87 @@
+defmodule Hangman.Game do
+  def new_game() do 
+    # function to filter alphabets
+    alphabets_only = fn x -> if Regex.match?(~r{[a-zA-Z]}, x),
```
Comment (pragdave) [inline hangman/lib/game/game.ex]: What's the difference between

~~~ elixir
if Regex.match?(~r{[a-zA-Z]}, x),
 do: true,
 else: false
~~~

and

~~~ elixir
Regex.match?(~r{[a-zA-Z]}, x)
~~~

File: hangman/lib/game/game.ex [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      turns_left: 7,
+      used: [],
+      last_guess: "",
+      letters: letters,
+      word: word
+    }
+  end
+
+  def tally(game) do
+    game
```
Comment (pragdave) [inline hangman/lib/game/game.ex]: This is a dangerous technique. What happens if you added a couple of secret fields to the struct. If you then forgot to update this code, you'd end up passing them to the client.

File: hangman/lib/game/game.ex [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    if Enum.member?(game.used, guess) do
+      game = %State{ game |
+        game_state: :already_guessed
+      }
+      { game, tally(game) }
+    else
+      make_move(game, guess, :new_guess)
+    end
+  end
+  def make_move(game, guess, _new_guess) do
```
Comment (pragdave) [inline hangman/lib/game/game.ex]: If I were grading this, the size and complexity of this function would lose you marks.

Try to keep your functions to just a couple of lines, and think of ways of avoiding all the `if` statements. It takes a little more thinking up front, but results in code that "just works".

The trick it to ask yourself "what one thing does this function do?" Write that, in terms of other functions. Then, for each of those, ask "what does this do."

**Comment by pragdave** [comment]
This breaks my heart—you seem to have a working application, but you were an hour late.

I'm going to break my own rules, just this once.

If you go through this code and tidy up the functions in Game to make them more functional, check it in and send me an email, I'll grade it, with a 10 point penalty. I'll need to see it by Sunday, 23:59.

I know you can make this tidier. Show me.

**Comment by ADi2991** [comment]
Thank you so much for your faith in me. I will do my best. I have one small
request though:
I know my code quality is far from perfect. I wish to learn from your
experience. Kindly grade my assignments as tough as you can. I could use
the nitpicking.

**Comment by ADi2991** [comment]
I have tried my best and revised the code quite a bit. Hope you like it.
Thanks again :)

**Pull request closed without merging**